File: December - 11/KishoreMuruganantham_counting_primes.swift
import Foundation

if let line = readLine() {
    var s = line
    if let r = s.range(of: ":") {
        s = String(s[r.upperBound...])
    }
    if let n = Int(s.trimmingCharacters(in: .whitespaces)) {
        var count = 0
        if n > 2 {
            var isPrime = [Bool](repeating: true, count: n)
            isPrime[0] = false
            isPrime[1] = false
            for i in 2..<n {
                if isPrime[i] {
                    count += 1
                    if i * i < n {
                        var j = i * i
                        while j < n {
                            isPrime[j] = false
                            j += i
                        }
                    }
                }
            }
        }
        print("The count of prime numbers less than \(n) is: \(count)")
    }
}

File: December - 11/KishoreMuruganantham_counting_primes.R
input <- file("stdin", "r")
line <- readLines(input, n = 1)
n_str <- sub(".*Enter N:\\s*", "", line)
n <- as.integer(n_str)

count <- 0
if (n > 2) {
  limit <- n - 1
  is_prime <- rep(TRUE, limit)
  is_prime[1] <- FALSE
  
  max_i <- floor(sqrt(limit))
  for (i in 2:max_i) {
    if (is_prime[i]) {
      mults <- seq(i * i, limit, by = i)
      is_prime[mults] <- FALSE
    }
  }
  count <- sum(is_prime)
}

cat("The count of prime numbers less than ", n, " is: ", count, "\n", sep = "")
